index.bs

```diff
@@ -6445,6 +6455,116 @@ The [=remote end steps=] with |command parameters| and |session| are:
+      }
+
+      emulation.SetScreenSettingsOverrideParameters = {
+        screenArea: emulation.ScreenArea / null,
+        ? contexts: [+browsingContext.BrowsingContext],
+        ? userContexts: [+browser.UserContext],
+      }
+    </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+      <pre class="cddl" data-cddl-module="local-cddl">
+      emulation.SetScreenSettingsOverrideResult = EmptyResult
+      </pre>
+   </dd>
+</dl>
+
+<div algorithm>
+
+The <dfn export>WebDriver BiDi emulated screen area</dfn> steps given a given [=/navigable=] |navigable|:
+
+1. Let |top-level traversable| be |navigable|’s [=navigable/top-level traversable=].
+
+1. Let |user context| be |top-level traversable|'s [=associated user context=].
+
+1. If [=screen area overrides map=] [=map/contains=] |top-level traversable|, return
+   [=screen area overrides map=][|top-level traversable|].
+
+1. If [=screen area overrides map=] [=map/contains=] |user context|, return
+   [=screen area overrides map=][|user context|].
+
+1. Return null
+
+</div>
+
+<div algorithm="remote end steps for emulation.setScreenSettingsOverride">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. If |command parameters| [=map/contains=] "<code>userContexts</code>"
+   and |command parameters| [=map/contains=] "<code>contexts</code>",
+   return [=error=] with [=error code=] [=invalid argument=].
+
+1. If |command parameters| doesn't [=map/contain=] "<code>userContexts</code>"
+   and |command parameters| doesn't [=map/contain=] "<code>contexts</code>",
+   return [=error=] with [=error code=] [=invalid argument=].
+
+1. Let |emulated screen area| be |command parameters|["<code>screenArea</code>"].
+
+1. If |emulated screen area| is not null:
+
+   1. [=map/Set=] |emulated screen area|["<code>x</code>"] to 0.
+
+   1. [=map/Set=] |emulated screen area|["<code>y</code>"] to 0.
+
+1. Let |navigables| be a [=/set=].
+
+1. If the <code>contexts</code> field of |command parameters| is present:
+
+   1. Let |navigables| be the result of [=trying=] to
+      [=get valid top-level traversables by ids=] with
+      |command parameters|["<code>contexts</code>"].
+
+1. Otherwise:
+
+   1. Assert the <code>userContexts</code> field of |command parameters| is present.
+
+   1. Let |user contexts| be the result of [=trying=] to [=get valid user contexts=]
+      with |command parameters|["<code>userContexts</code>"].
+
+   1. For each |user context| of |user contexts|:
+
+      1. If |emulated screen area| is null, [=map/remove=] |user context| from
+         [=screen area overrides map=].
+
+      1. Otherwise, [=map/set=] [=screen area overrides map=][|user context|] to |emulated screen area|.
+
+      1. [=list/For each=] |top-level traversable| of the list of all
+         [=/top-level traversables=] whose [=associated user context=] is
+         |user context|:
+
+         1. [=list/Append=] |top-level traversable| to |navigables|.
+
+1. For each |navigable| of |navigables|:
```

Comment:
The config per navigable should be stored in the `1. If the <code>contexts</code> field of |command parameters| is present:` case, as otherwise setting `null` to user context would remove per-navigable emulations.